typing/types.mli

```diff
@@ -700,6 +701,7 @@ type label_description =
     lbl_res: type_expr;                 (* Type of the result *)
     lbl_arg: type_expr;                 (* Type of the argument *)
     lbl_mut: mutable_flag;              (* Is this a mutable field? *)
+    lbl_atomic: bool;                   (* Is this an atomic field? *)
```

Comment:
My guess is that adding `Atomic` as a third value would make the code a mess, because it entangles two aspect that are more orthogonal than that. (It is not *useful* to have immutable atomics, and we cannot express them without adding more type-level information, but it is certainly something that makes sense.)

Reply:
I think that's fair.

Reply:
> (It is not _useful_ to have immutable atomics, and we cannot express them without adding more type-level information, but it is certainly something that makes sense.)

I don't think it makes sense to have immutable atomics at all. Immutable values have semantics that are completely incompatible with atomic behaviour. I'm happy with keeping two flags though: it's a bit of a waste of memory but there are lots of cases where mutable values are treated the same, atomic or not.